lib/propolis/src/vcpu.rs

```diff
@@ -33,6 +35,18 @@ pub const MAXCPU: usize = bhyve_api::VM_MAXCPU;
 #[cfg(feature = "omicron-build")]
 pub const MAXCPU: usize = 64;
 
+#[derive(Debug, Error)]
+pub enum GetCpuidError {
+    #[error("failed to read CPUID values from bhyve")]
+    ReadIoctlFailed(#[source] std::io::Error),
+
+    #[error("failed to build a map of CPUID entries")]
+    MapConversion(#[from] CpuidMapConversionError),
+
+    #[error("failed to convert CPUID values to vendor string: {0}")]
+    VendorConversionFailed(&'static str),
```

Comment:
should this maybe be reworded to something like
```suggestion
    #[error("unrecognized CPUID vendor string: {0}")]
    VendorConversionFailed(&'static str),
```

because, technically, we _did_ convert it to a vendor string, but the vendor string was just, like, "CyrixInstead" or some other weird thing that we don't care to support...